tap_exchangeratesapi/__init__.py

```diff
@@ -70,7 +75,38 @@ def do_sync(base, start_date):
                 singer.write_schema('exchange_rate', schema, 'date')
 
             if payload['date'] == next_date:
-                singer.write_records('exchange_rate', [parse_response(payload)])
+                if catalog_override:
+                    catalog_stream_override = [
+                        x for x in catalog_override["streams"]
+                        if x["tap_stream_id"] == "exchange_rate"
+                    ]
+                    if not catalog_stream_override:
+                        raise ValueError(
+                            "Stream 'exchange_rate' not found in "
+                            f"json: {catalog_override}"
+                        )
+                    # else:
+                    #     logger.info(catalog_stream_override)
+                    metadata_override = catalog_stream_override[0]["metadata"]
+                    if not metadata_override:
+                        raise ValueError(
+                            "Metadata not found in "
+                            f"json: {catalog_override}"
+                        )
+                    # else:
+                    #     logger.info(metadata_override)
+                    logger.info("Replicating with provided catalog file override")
+                    for record in [parse_response(payload)]:
+                        singer.write_record(
+                            'exchange_rate',
+                            singer.Transformer().transform(
+                                data=record,
+                                schema=catalog_stream_override[0]["schema"],
+                                # metadata=metadata_override, # TODO: debug (not working)
```

Comment:
I get errors when I try to pass the `metadata` arg. 

- [ ] Any ideas?? ❓

Reply:
@dmosorast 

I think you are correct. The below seems to resolve the error, now pulling in the full `dict` which is one level up from the "metadata" element (which is itself of type `list`).

```
                            singer.Transformer().transform(
                                data=record,
                                schema=catalog_stream_override[0]["schema"],
                                metadata=catalog_stream_override[0],
                            )
```

Reply:
More to follow (and more code cleanup to do) once I confirm the results are as expected.

Reply:
I think the behavior might be incorrect. Without 100% tracing the code, I would expect it to be something like `metadata.to_map(metadata_override)` using the `metadata` module in singer-python.

Reply:
The API for the metadata is a bit strange since it uses a module/"functional" pattern, so I can chat over DM if anything needs cleared up. I'm trying to figure out how to explain it myself.